http-client/src/main/java/io/airlift/http/client/jetty/BufferingResponseListener.java

```diff
@@ -126,10 +121,10 @@ public String getEncoding()
      */
     public byte[] getContent()
     {
-        if (buffer == null) {
+        if (buffer.getLength() == 0) {
             return new byte[0];
         }
-        return BufferUtil.toArray(buffer);
+        return buffer.takeByteBuffer().array();
```

Comment:
In case you missed my comment from the old PR: https://github.com/airlift/airlift/pull/1349#discussion_r1913932700

Reply:
This is a sized copy so always capacity == remaining